libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,452 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  coerceBooleanProperty,
+  coerceNumberProperty,
+} from '@angular/cdk/coercion';
+import {
+  DOWN_ARROW,
+  END,
+  HOME,
+  LEFT_ARROW,
+  PAGE_DOWN,
+  PAGE_UP,
+  RIGHT_ARROW,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import { Platform } from '@angular/cdk/platform';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  clamp,
+  isDefined,
+  roundToDecimal,
+} from '@dynatrace/barista-components/core';
+import { DtInput } from '@dynatrace/barista-components/input';
+import {
+  animationFrameScheduler,
+  BehaviorSubject,
+  fromEvent,
+  merge,
+  Observable,
+  of,
+  Subject,
+} from 'rxjs';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  filter,
+  map,
+  observeOn,
+  share,
+  switchMap,
+  takeUntil,
+  withLatestFrom,
+} from 'rxjs/operators';
+import {
+  getKeyCodeValue,
+  getSliderPositionBasedOnValue,
+  getSliderValueForCoordinate,
+  roundToSnap,
+} from './slider-utils';
+
+/** We need unique ids in order to have correct labeling. */
+let uniqueId = 0;
+
+const KEY_CODES_ARRAY: Array<number> = [
+  LEFT_ARROW,
+  DOWN_ARROW,
+  RIGHT_ARROW,
+  UP_ARROW,
+  HOME,
+  END,
+  PAGE_UP,
+  PAGE_DOWN,
+];
+
+declare const window: any;
+
+@Component({
+  selector: 'dt-slider',
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.scss'],
+  host: {
+    class: 'dt-slider',
+  },
+  encapsulation: ViewEncapsulation.Emulated,
+  exportAs: 'dtSlider',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtSlider implements AfterViewInit, OnDestroy, OnInit {
+  /**
+   * Holds the value, the rounding is shifted with.
+   * Calculated based in the step, to avoid JS rounding problems.
+   */
+  private _roundShift: number = 0;
+
+  /** @internal Unique id for this input. */
+  _labelUid = `dt-slider-label-${uniqueId++}`;
```

Comment:
Maybe align with @tomheller and think about a dedicated mixin for unique id's we have this behaviour now quite often.

Reply:
Good idea, sounds like a good future improvement.